src/data_readers/data_reader_jag_conduit.cpp

```diff
@@ -925,10 +862,18 @@ if (is_master()) std::cout << "SETTING m_is_data_loaded = true\n";
   select_subset_of_data();
 }
 
+void data_reader_jag_conduit::preload_helper(const hid_t& h, const std::string &sample_name, const std::string &field_name, int data_id, conduit::Node &node) {
+  const std::string path = sample_name + field_name;
+  if (!has_path(h, path)) {
+    LBANN_ERROR("!has_path(h, path) for path: ", path);
+  }
+  conduit::Node work;
```

Comment:
This is going to copy the value from the conduit node I think that you want something like:
` const std::string key2 = '/' + LBANN_DATA_ID_STR(data_id) + field_name;
   conduit::Node& work = node[key2];`

Reply:
Or even `const std::string key2 = build_string('/', LBANN_DATA_ID_STR(data_id), field_name);` 😃

Reply:
That block looks for the path in the conduit node on disk; 'h' is hid_t.  The node has not been read into memory at that point. We could eliminate those three lines, because if the path doesn't exist conduit will throw an exception. That would avoid the cost of the check.

Reply:
> build_string

Looks like that instantiates extra objects: ostringstream, then returns a copy of the ss.str(). Of course, I don't know how smart the compiler might be about optimizing that.

Reply:
If this is not a performance-critical section, it won't matter, and the code will be more understandable. My guess is that the cost of creating an ostringstream, piping things into it, and copying the string back out will be significantly faster than the `read_node` call... That said, this is not a compelling use-case for `build_string` since, seemingly, everything is either a string or convertible to it. (`build_string` really exists to simplify `std::string s = std::to_string(some_var) + " " + std::to_string(other_var) ... ;` use-cases in non-performance-critical code sections.)